app/src/main/java/org/mozilla/fenix/onboarding/view/UpgradeOnboardingScreen.kt

```diff
@@ -0,0 +1,215 @@
+        ) {
+            Icon(
+                painter = painterResource(id = R.drawable.mozac_ic_close),
+                contentDescription = stringResource(R.string.onboarding_home_content_description_close_button),
+                tint = FirefoxTheme.colors.iconPrimary,
+            )
+        }
+
+        OnboardingPage(
+            onboardingPageUiState = when (onboardingState) {
+                UpgradeOnboardingState.Welcome -> OnboardingPageUiState(
+                    image = R.drawable.ic_onboarding_welcome,
+                    title = R.string.onboarding_home_welcome_title_2,
+                    description = R.string.onboarding_home_welcome_description,
+                    primaryButtonText = R.string.onboarding_home_get_started_button,
+                    recordImpressionEvent = { Onboarding.welcomeCardImpression.record(NoExtras()) },
+                )
+                UpgradeOnboardingState.SyncSignIn -> OnboardingPageUiState(
+                    image = R.drawable.ic_onboarding_sync,
+                    title = R.string.onboarding_home_sync_title_3,
+                    description = R.string.onboarding_home_sync_description,
+                    primaryButtonText = R.string.onboarding_home_sign_in_button,
+                    secondaryButtonText = R.string.onboarding_home_skip_button,
+                    recordImpressionEvent = { Onboarding.syncCardImpression.record(NoExtras()) },
+                )
+            },
+            onPrimaryButtonClick = onPrimaryButtonClick,
+            onSecondaryButtonClick = onSecondaryButtonClick,
+            modifier = Modifier
+                .weight(1f)
+                .fillMaxWidth()
+                .padding(horizontal = 16.dp)
+                .verticalScroll(rememberScrollState()),
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        Indicators(
+            onboardingState = onboardingState,
+        )
+    }
+}
+
+@Composable
+private fun Indicators(onboardingState: UpgradeOnboardingState) {
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.Center,
+    ) {
+        Indicator(
+            color = if (onboardingState == UpgradeOnboardingState.Welcome) {
+                FirefoxTheme.colors.indicatorActive
+            } else {
+                FirefoxTheme.colors.indicatorInactive
+            },
+        )
+
+        Spacer(modifier = Modifier.width(8.dp))
+
+        Indicator(
+            color = if (onboardingState == UpgradeOnboardingState.SyncSignIn) {
+                FirefoxTheme.colors.indicatorActive
+            } else {
+                FirefoxTheme.colors.indicatorInactive
+            },
+        )
+    }
+}
+
+@Composable
+private fun Indicator(color: Color) {
+    Box(
+        modifier = Modifier
+            .size(6.dp)
+            .clip(CircleShape)
+            .background(color),
+    )
+}
+
+@Composable
+@Preview
+private fun PreviewUpgradeOnboardingScreen() {
```

Comment:
```suggestion
private fun UpgradeOnboardingScreenPreview() {
```
This is just to stick with the Compose Preview naming convention.

Reply:
I don't expect this to work nor do I think we need this.

Reply:
Hmm yeah, the idea was to start the function with Preview so it doesn't pollute autocomplete, let me know what you think? 
If there's a convention, can i read it somewhere?

Reply:
Btw the preview should work, I think maybe because metrics are now only called in side effects (click listeners and effect handlers), and they are not in the composition.

Reply:
We have mostly adopted `Preview` being the suffix everywhere because that's how it is in the Compose Preview docs https://developer.android.com/jetpack/compose/tooling/studio#preview

Reply:
Because the preview functions are private, they won't pollute the global namespace.